app/lib/common/widgets/attachments/post_attachment_selection.dart

```diff
@@ -99,86 +92,57 @@ class _PostAttachmentSelectionState
-        } else if (mimeType.startsWith('audio/')) {
+              .size(bytes.length)
+              .width(decodedImage.width)
+              .height(decodedImage.height);
+          final attachmentDraft = await manager.contentDraft(imageDraft);
+          drafts.add(attachmentDraft);
+        } else if (type == AttachmentType.audio) {
+          Uint8List bytes = await file.readAsBytes();
           final audioDraft =
-              client.audioDraft(file.path, mimeType).size(file.lengthSync());
-          if (inputState.repliedToMessage != null) {
-            await stream.replyMessage(
-              inputState.repliedToMessage!.id,
-              audioDraft,
-            );
-          } else {
-            await stream.sendMessage(audioDraft);
-          }
-        } else if (mimeType.startsWith('video/')) {
+              client.audioDraft(file.path, mimeType).size(bytes.length);
+          final attachmentDraft = await manager.contentDraft(audioDraft);
+          drafts.add(attachmentDraft);
+        } else if (type == AttachmentType.video) {
+          Uint8List bytes = await file.readAsBytes();
           final videoDraft =
-              client.videoDraft(file.path, mimeType).size(file.lengthSync());
-          if (inputState.repliedToMessage != null) {
-            await stream.replyMessage(
-              inputState.repliedToMessage!.id,
-              videoDraft,
-            );
-          } else {
-            await stream.sendMessage(videoDraft);
-          }
+              client.videoDraft(file.path, mimeType).size(bytes.length);
+          final attachmentDraft = await manager.contentDraft(videoDraft);
+          drafts.add(attachmentDraft);
         } else {
-          final draft =
-              client.fileDraft(file.path, mimeType).size(file.lengthSync());
-          if (inputState.repliedToMessage != null) {
-            await stream.replyMessage(inputState.repliedToMessage!.id, draft);
-          } else {
-            await stream.sendMessage(draft);
-          }
+          String fileName = file.path.split('/').last;
+          final fileDraft = client
+              .fileDraft(file.path, mimeType)
+              .filename(fileName)
+              .size(file.lengthSync());
+          final attachmentDraft = await manager.contentDraft(fileDraft);
+          drafts.add(attachmentDraft);
         }
       }
-    } catch (e, s) {
-      _log.severe('error occurred sending attachments', e, s);
-    }
-
-    if (inputState.repliedToMessage != null) {
-      final notifier = ref.read(chatInputProvider(roomId).notifier);
-      notifier.setRepliedToMessage(null);
-      notifier.setEditMessage(null);
-      notifier.showReplyView(false);
-      notifier.showEditView(false);
-      notifier.setReplyWidget(null);
-      notifier.setEditWidget(null);
+      for (var draft in drafts) {
+        final res = await draft.send();
+        _log.info('attachment sent: $res');
+      }
+    } catch (e) {
+      EasyLoading.dismiss();
```

Comment:
we should show that error to the user, too!
```suggestion
      EasyLoading.error('Error sending attachments', e);
```